pupil-spa/src/app/services/answer/answer.service.ts

```diff
@@ -15,8 +18,24 @@ export class AnswerService {
       answers = [];
     }
 
+    // Check for duplicates - we can only store one answer for each question
+    const isAlreadyAnswered = answers.filter(
+      x => {
+        if (x.sequenceNumber === answer.sequenceNumber
+        && x.factor1 === answer.factor1
+        && x.factor2 === answer.factor2) {
+          return true;
+        }
+        return false;
+      });
+
+    if (isAlreadyAnswered.length) {
```

Comment:
will this not always be initialised to zero or more?

Reply:
It's the result of a `filter` so will be an array.  If it finds any there will an element in the array.  Perhaps the name suggests it is a boolean when it isn't.  What do you suggest?

Reply:
I agree that this sounds like a boolean. `alreadyAnswered` would work nicely.